### PR TITLE
TF-3084 Handle drop image to identity editor for public asset

### PR DESCRIPTION
**Pull request by tddang-linagora**
## Issue
- #3084 

## Demo

https://github.com/user-attachments/assets/55678f05-7ab4-45ae-a97e-d5b9f1d8c9c3

### CODE REVIEW

File: lib/features/identity_creator/presentation/identity_creator_controller.dart [hunk 1]
```diff
@@ -701,4 +769,31 @@ class IdentityCreatorController extends BaseController {
...
+
+    if (listImages.isEmpty && listFileInfo.isNotEmpty && context.mounted) {
+      appToast.showToastErrorMessage(
+        context,
+        AppLocalizations.of(context).canNotUploadFileToSignature
+      );
+      return;
+    }
+
+    listImages
```
Comment (dab246) [inline lib/features/identity_creator/presentation/identity_creator_controller.dart:792]: please check validate maximum size

File: lib/features/identity_creator/presentation/identity_creator_controller.dart [hunk 2]
```diff
@@ -701,4 +769,31 @@ class IdentityCreatorController extends BaseController {
...
+      .map((fileInfo) => fileInfo.toPlatformFile())
+      .forEach((platformFile) => _insertInlineImage(
```
Comment (dab246) [inline lib/features/identity_creator/presentation/identity_creator_controller.dart:794]: Please use `for` or `Future.forEach`  to handle async

Comment (tddang-linagora) [reply lib/features/identity_creator/presentation/identity_creator_controller.dart]: Done

File: lib/features/identity_creator/presentation/identity_creator_controller.dart (hunk 1 above)
Comment (tddang-linagora) [reply lib/features/identity_creator/presentation/identity_creator_controller.dart]: Done

File: lib/features/composer/presentation/widgets/web/local_file_drop_zone_widget.dart [hunk 3]
```diff
@@ -52,7 +52,7 @@ class LocalFileDropZoneWidget extends StatelessWidget {
               child: Column(
                 mainAxisSize: MainAxisSize.min,
                 children: [
-                  SvgPicture.asset(imagePaths.icDropZoneIcon),
+                  Expanded(child: SvgPicture.asset(imagePaths.icDropZoneIcon)),
```
Comment (dab246) [inline lib/features/composer/presentation/widgets/web/local_file_drop_zone_widget.dart:55]: Does it affect the display on composer? Please check again.

Comment (tddang-linagora) [reply lib/features/composer/presentation/widgets/web/local_file_drop_zone_widget.dart]: It does affect somewhat. Updated

**Comment by dab246** [comment]
It still doesn't work. Please see below demo video:




https://github.com/user-attachments/assets/a94fd55b-3519-46a7-a178-f3432756bfb6

**Comment by dab246** [comment]
- [ ] We should add a `loading icon` when picking images.  (Server Prod - Debug Mode)


https://github.com/user-attachments/assets/dda64d58-7e54-440f-94d8-f7f7035e0557

- [ ] The width of the image exceeds the width of the identity creator screen.



Screenshot 2024-08-16 at 11 11 07

**Comment by tddang-linagora** [comment]
>It still doesn't work.
>We should add a loading icon when picking images.
>The width of the image exceeds the width of the identity creator screen.

Done

https://github.com/user-attachments/assets/3f66ea86-79bd-4f5d-8d0c-5fbeb23cb12a

**Comment by dab246** [comment]
> > It still doesn't work.
> > We should add a loading icon when picking images.
> > The width of the image exceeds the width of the identity creator screen.
> 
> Done
> 
>  Screen.Recording.2024-08-16.at.14.21.02.mov

I fetch latest code but it still error 


https://github.com/user-attachments/assets/bc0a0f88-8480-4408-bbff-3f5396068360

**Comment by hoangdat** [comment]
- [ ] after drag n drop, text in name is selected (like ctrl + A)

**Comment by hoangdat** [comment]
- [ ] copy/paste image still in base 64 (create another ticket to handle it), but now, we should not show base64 image inside editor

**Comment by hoangdat** [comment]
- [ ] when I tried to drag n drop 2 different images to identity, but only the first one image display but duplicate in identity 

https://github.com/user-attachments/assets/64f09c41-39f9-4c68-b3f2-2e392fde7332

**Comment by tddang-linagora** [comment]
> * [ ]  copy/paste image still in base 64 (create another ticket to handle it), but now, we should not show base64 image inside editor

How should we handle it? Transform it to public asset if user copy/paste base64?

**Comment by tddang-linagora** [comment]
>when I tried to drag n drop 2 different images to identity, but only the first one image was displayed but duplicate in identity

Please try again

**Comment by tddang-linagora** [comment]
>I fetch latest code but it still error

I cannot reproduce the bug. Please provide steps.

**Comment by tddang-linagora** [comment]
>after drag n drop image to identity, text in name is selected (like ctrl + A)

can't fix. it's a Flutter bug and happening app wide.

**Comment by dab246** [comment]
> > I fetch latest code but it still error
> 
> I cannot reproduce the bug. Please provide steps.

Please see below video demo:


https://github.com/user-attachments/assets/ab116ed5-faec-4aaf-9f5f-608e0399d2e8

**Comment by dab246** [comment]
> > > I fetch latest code but it still error
> > 
> > 
> > I cannot reproduce the bug. Please provide steps.
> 
> Please see below video demo:
> 
>  demo.mov

Fixed. 


https://github.com/user-attachments/assets/3052b1cc-a940-409e-8ad1-d2b6ed2f950b

**Review by dab246 — APPROVED** [review]
LGTM

File: lib/features/identity_creator/presentation/identity_creator_controller.dart [hunk 4]
```diff
@@ -218,7 +288,7 @@ class IdentityCreatorController extends BaseController {
     }
   }
 
-  void _getALlIdentitiesSuccess(GetAllIdentitiesSuccess success) {
+  void _getAllIdentitiesSuccess(GetAllIdentitiesSuccess success) {
```
[the remark this answers is not in the document]
Comment (dab246) [reply lib/features/identity_creator/presentation/identity_creator_controller.dart:291]: why? It is currently spelled incorrectly, we need to fix it.

**Review by hoangdat — APPROVED** [review]

**Pull request merged**